src/nnpack/nnlib.jl

```diff
@@ -0,0 +1,86 @@
+const AA{N} = AbstractArray{Float32,N}
```

Comment:
I think that we should catch Float64 here as well, at least for convs and possibly also for softmax and pooling (depending on benchmark results). Doing the conversion to Float32 and feeding to nnpack is probably quicker than  using our slow convolutions.

Reply:
This will definitely make things faster but we also end up losing features like dilation, also arbitrary stride kernel combinations will give errors. Also if we are internally making all computations in Float32 can't we simply make the conversion in Flux, we would just need one conversion in that case instead of repeated conversion back and forth from Float64 -> Float32

Reply:
If we're calling into `NNPACK`, I think it makes sense to "poison" the chain of operations; e.g. if we've configured Flux to be dead-set on using NNPACK, then `conv(X<:Float64, W<:Float32)` should return a `Float32`; narrowing the output type due to the internal kernel working in f32.  This should cause downstream operations to use `conv(X<:Float32, W<:Float32)`, thereby avoiding the constant re-quantization.

If we just document this, and allow people who really care about precision to trivially re-cast to `Float64` in the places that it matters, then I think we should be able to get the best of both worlds.

As for dilation and arbitrary stride kernels, I think those should fall back to the slow convolution algorithms.

Reply:
The idea to document it is fair enough.
However, I am not yet sure how to call those functions for dilation, now that they are overloaded.

Reply:
Perhaps you can use `invoke`?